File: validarCPF/testarCPF.py
def testaCPF( cpf):
   
    soma = 0

    cpf = retiraCaracteresInvalidos(cpf);

    if len(cpf) != 11:
        return False

    if (allCharactersSame(cpf)):
        return False

    for i in range(1, 10):
        soma += int(cpf[i - 1]) * (11 - i)

    resto = (soma * 10) % 11

    if (resto == 10) or (resto == 11):
        resto = 0

    if (resto != int(cpf[9])):
        return False

    soma = 0

    for i in range(1, 11):
        soma += int(cpf[i - 1]) * (12 - i)

    resto = (soma * 10) % 11

    if (resto == 10) or (resto == 11):
        resto = 0

    if (resto != int(cpf[10])):
        return False

    return True


def retiraCaracteresInvalidos(string):
    string = string.replace(".", "")
    string = string.replace("-", "")
    string = string.replace(" ", "")
    return string


def allCharactersSame(string):
    for char in string:
        if (char == string[0]):
            return False
    return True
    

File: validarCPF/testarCPF.java
package validarCPF;


public class testarCPF {

    public static boolean testaCPF(String cpf){
        double soma = 0;
        double resto;

        cpf = retiraCaracteresInvalidos(cpf);

        if (cpf.length() != 11){
            return false;
        }

        if (allCharactersSame(cpf)) {
            return false;
        }

        for (int i=1; i<=9; i++) {
            soma += Integer.parseInt(Character.toString(cpf.charAt(i-1))) * (11 - i);
        }
        
        resto = (soma * 10) % 11;
        if ((resto == 10) || (resto == 11)) {
            resto = 0;
        }

        if (resto != Integer.parseInt(Character.toString(cpf.charAt(9)))) {
            return false;
        }

        soma = 0;

        for (int i=1; i<=10; i++) {
            soma += Integer.parseInt(Character.toString(cpf.charAt(i-1))) * (12 - i);
        }

        resto = (soma * 10) % 11;
        if ((resto == 10) || (resto == 11)) {
            resto = 0;
        }

        if (resto != Integer.parseInt(Character.toString(cpf.charAt(10)))) {
            return false;
        }

        return true;
    }


    public static String retiraCaracteresInvalidos(String s){
        s = s.replace(".", "");
        s = s.replace("-", "");
        s = s.replace(" ", "");
        return s;
    }


    static boolean allCharactersSame(String s){
        for (int i = 1; i < s.length(); i++)
            if (s.charAt(i) != s.charAt(0))
                return false;
        return true;
    }
}
